### PR TITLE
feat: Add more actions

### PR DESCRIPTION
**Pull request by teolemon**
### What
- Add more actions to panels
  - [x] start by listing all the actions that are possible in the app
  - [x] Add matching string in the GetText files
  - [x] Packaging  - templates/api/knowledge-panels/environment/packaging_recycling.tt.json
    - [ ] in the related the sub-knowledge panel
    - [ ] in the matching ecoscore sub-knowledge panel if an eco-score is already computed
    - [ ] Add differentiated support for packaging photo and packaging text
  - [x] Origins of ingredients - templates/api/knowledge-panels/environment/origins_of_ingredients.tt.json
    - [ ] in case there are none, in the Eco-Score sub-knowledge panel section about origins if an eco-score is already computed

### CODE REVIEW

File: templates/api/knowledge-panels/environment/origins_of_ingredients.tt.json [hunk 1]
```diff
@@ -12,6 +12,15 @@
         "icon_color_from_evaluation": true,
         "icon_url": "[% static_subdomain %]/images/icons/dist/public.svg",
     },
+        {
+            "element_type": "action",
+            "action_element": {
+                "html": `[% lang("actions_add_origins") %]`,
```
Comment (teolemon) [inline templates/api/knowledge-panels/environment/origins_of_ingredients.tt.json:18]:
```suggestion
                "html": `[% lang("actions_origins") %]`,
```

File: po/common/common.pot [hunk 2]
```diff
@@ -6198,6 +6198,50 @@ msgctxt "action_add_nutrition_facts"
...
+
+msgctxt "action_ingredients_image"
+msgid "Take a photo of the ingredients"
+msgstr "Take a photo of the ingredients"
+
+msgctxt "action_nutrition_image"
+msgid "Take a photo of the nutrition facts"
+msgstr "Take a photo of the nutrition facts"
+
+msgctxt "action_packaging_photo"
```
Comment (teolemon) [inline po/common/common.pot:6217]:
```suggestion
msgctxt "action_packaging_image"
```

File: po/common/common.pot [hunk 3]
```diff
@@ -6198,6 +6198,50 @@ msgctxt "action_add_nutrition_facts"
...
+
+msgctxt "action_packaging_text"
+msgid "Extract and check the recycling information"
+msgstr "Extract and check the recycling information"
+
+msgctxt "action_stores"
+msgid "Add the store where you found this product"
+msgstr "Add the store where you found this product"
+
+msgctxt "action_origins"
```
Comment (teolemon) [inline po/common/common.pot:6229]:
```suggestion
msgctxt "action_add_origins"
```

File: po/common/common.pot [hunk 4]
```diff
@@ -6198,6 +6198,50 @@ msgctxt "action_add_nutrition_facts"
...
+
+msgctxt "action_packaging_photo"
+msgid "Take a photo of the recycling information"
+msgstr "Take a photo of the recycling information"
+
+msgctxt "action_packaging_text"
+msgid "Extract and check the recycling information"
+msgstr "Extract and check the recycling information"
+
+msgctxt "action_stores"
```
Comment (teolemon) [inline po/common/common.pot:6225]:
```suggestion
msgctxt "action_add_stores"
```

File: po/common/common.pot [hunk 5]
```diff
@@ -6198,6 +6198,50 @@ msgctxt "action_add_nutrition_facts"
...
+
+msgctxt "action_nutrition_image"
+msgid "Take a photo of the nutrition facts"
+msgstr "Take a photo of the nutrition facts"
+
+msgctxt "action_packaging_photo"
+msgid "Take a photo of the recycling information"
+msgstr "Take a photo of the recycling information"
+
+msgctxt "action_packaging_text"
```
Comment (teolemon) [inline po/common/common.pot:6221]:
```suggestion
msgctxt "action_add_packaging_text"
```

File: po/common/common.pot [hunk 6]
```diff
@@ -6198,6 +6198,50 @@ msgctxt "action_add_nutrition_facts"
...
+
+msgctxt "action_portion_size"
+msgid "Add portion size"
+msgstr "Add portion size"
+
+msgctxt "action_ingredients_image"
+msgid "Take a photo of the ingredients"
+msgstr "Take a photo of the ingredients"
+
+msgctxt "action_nutrition_image"
```
Comment (teolemon) [inline po/common/common.pot:6213]:
```suggestion
msgctxt "action_add_nutrition_image"
```

File: po/common/common.pot [hunk 7]
```diff
@@ -6198,6 +6198,50 @@ msgctxt "action_add_nutrition_facts"
...
 
+msgctxt "action_add_basic_details"
+msgid "Add basic details"
+msgstr "Add basic details"
+
+msgctxt "action_portion_size"
+msgid "Add portion size"
+msgstr "Add portion size"
+
+msgctxt "action_ingredients_image"
```
Comment (teolemon) [inline po/common/common.pot:6209]:
```suggestion
msgctxt "action_add_ingredients_image"
```

File: po/common/common.pot [hunk 8]
```diff
@@ -6198,6 +6198,50 @@ msgctxt "action_add_nutrition_facts"
 msgid "Add nutrition facts"
 msgstr "Add nutrition facts"
 
+msgctxt "action_add_basic_details"
+msgid "Add basic details"
+msgstr "Add basic details"
+
+msgctxt "action_portion_size"
```
Comment (teolemon) [inline po/common/common.pot:6205]:
```suggestion
msgctxt "action_add_portion_size"
```

File: templates/api/knowledge-panels/environment/origins_of_ingredients.tt.json [hunk 9]
```diff
@@ -12,6 +12,15 @@
         "icon_color_from_evaluation": true,
         "icon_url": "[% static_subdomain %]/images/icons/dist/public.svg",
     },
+        {
+            "element_type": "action",
+            "action_element": {
+                "html": `[% lang("actions_origins") %]`,
```
Comment (teolemon) [inline templates/api/knowledge-panels/environment/origins_of_ingredients.tt.json:18]:
```suggestion
                "html": `[% lang("action_add_origins") %]`,
```

File: templates/api/knowledge-panels/environment/packaging_recycling.tt.json [hunk 10]
```diff
@@ -34,6 +34,15 @@
                     `
             }
         },
+              {
+            "element_type": "action",
+            "action_element": {
+                "html": `[% lang("actions_add_packaging_image") %]`,
```
Comment (teolemon) [inline templates/api/knowledge-panels/environment/packaging_recycling.tt.json:40]:
```suggestion
                "html": `[% lang("action_add_packaging_image") %]`,
```

File: lib/ProductOpener/Products.pm [hunk 11]
```diff
@@ -2371,12 +2371,41 @@ sub product_action_url($code, $action) {
...
+		$url .= "#stores";
+	}
+	elsif ($action eq "add_packager_codes_image") {
+		$url .= "#packager_codes";
+	}
+	elsif ($action eq "add_labels") {
+		$url .= "#labels";
+	}
+	elsif ($action eq "add_countries") {
+		$url .= "#contries";
```
Comment (teolemon) [inline lib/ProductOpener/Products.pm:2406]:
```suggestion
		$url .= "#countries";
```

**Comment by M123-dev** [comment]
👍🏼

File: templates/api/knowledge-panels/health/ingredients/nova.tt.json [hunk 12]
```diff
@@ -54,6 +54,9 @@
 <p>&rarr; <a href="/nova">[% lang("nova_classification_learn_more") %]</a></p>
 `
             },        
-        },        
+        },
+[% IF product.nova_group_error = 'missing_ingredients' %]
+action
+[% ENDIF %]
```
Comment (teolemon) [inline templates/api/knowledge-panels/health/ingredients/nova.tt.json:60]:
```suggestion
        },
[% IF product.nova_group_error = 'missing_ingredients' %]
{
            "element_type": "action",
            "action_element": {
                "html": `[% lang("action_add_ingredients_image") %]`,
                "actions": [
                    "add_ingredients_image"
                ]
            }
        },   
[% ENDIF %]

```

File: lib/ProductOpener/Products.pm [hunk 13]
```diff
@@ -2497,12 +2497,41 @@ sub product_action_url ($code, $action) {
...
 		$url .= "#categories";
 	}
+	elsif ($action eq "add_ingredients_image") {
+		$url .= "#ingredients";
+	}
 	elsif ($action eq "add_ingredients_text") {
 		$url .= "#ingredients";
 	}
+	elsif ($action eq "add_nutrition_facts_image") {
+		$url .= "#nutrition;
```
Comment (teolemon) [inline lib/ProductOpener/Products.pm:2507]:
```suggestion
		$url .= "#nutrition";
```

File: lib/ProductOpener/Products.pm [hunk 14]
```diff
@@ -2497,12 +2497,41 @@ sub product_action_url ($code, $action) {
...
 	elsif ($action eq "add_nutrition_facts") {
 		$url .= "#nutrition";
 	}
+	elsif ($action eq "add_packaging_image") {
+		$url .= "#packaging";
+	}
+	elsif ($action eq "add_packaging_text") {
+		$url .= "#packaging";
+	}
+	# will do nothing unless a custom section is added
```
Comment (teolemon) [inline lib/ProductOpener/Products.pm:2518]:
```suggestion
	# Note: 27/11/2022 - The following HTML anchors will do nothing unless a custom HTML anchor is added in the future
```

File: lib/ProductOpener/Products.pm [hunk 15]
```diff
@@ -2497,12 +2497,41 @@ sub product_action_url ($code, $action) {
...
 	elsif ($action eq "add_nutrition_facts") {
 		$url .= "#nutrition";
 	}
+	elsif ($action eq "add_packaging_image") {
+		$url .= "#packaging";
+	}
+	elsif ($action eq "add_packaging_text") {
+		$url .= "#packaging";
+	}
+	# Note: 27/11/2022 - The following HTML anchors will do nothing unless a custom HTML anchor is added in the future
```
Comment (teolemon) [inline lib/ProductOpener/Products.pm:2518]:
```suggestion
	# Note: 27/11/2022 - Pierre - The following HTML anchors will do nothing unless a custom HTML anchor is added in the future
```

File: lib/ProductOpener/Products.pm [hunk 16]
```diff
@@ -2497,12 +2497,41 @@ sub product_action_url ($code, $action) {
...
 	elsif ($action eq "add_nutrition_facts") {
 		$url .= "#nutrition";
 	}
+	elsif ($action eq "add_packaging_image") {
+		$url .= "#packaging";
+	}
+	elsif ($action eq "add_packaging_text") {
+		$url .= "#packaging";
+	}
+	# Note: 27/11/2022 - Pierre - The following HTML anchors will do nothing unless a custom HTML anchor is added in the future
```
Comment (teolemon) [inline lib/ProductOpener/Products.pm:2518]:
```suggestion
	# Note: 27/11/2022 - Pierre - The following HTML anchors will do nothing unless a custom HTML anchor is added in the future to the product edition template
```

**Comment by teolemon** [comment]
#   Failed test at tests/unit/templates.t line 42.
# failed to fetch template: file error - parse error - api/knowledge-panels/health/ingredients/nova.tt.json line 58: unexpected token (=)
#   [% IF product.nova_group_error = 'missing_ingredients' %]

**Comment by M123-dev** [comment]
Is the problem fixed by now

**Comment by teolemon** [comment]
No, we need to generate new test outputs for the build to go green, and the PR to be mergeable

File: templates/api/knowledge-panels/environment/origins_of_ingredients.tt.json [hunk 17]
```diff
@@ -12,6 +12,15 @@
...
     },
+        {
+            "element_type": "action",
+            "action_element": {
+                "html": `[% lang("action_add_origins") %]`,
+                "actions": [
+                    "add_origins",
+                ]
+            }
+        },
```
Comment (alexgarel) [inline templates/api/knowledge-panels/environment/origins_of_ingredients.tt.json:23]: You broke the json here ;-) (it should be included after "elements")
I'm fixing it.

**Review by alexgarel — APPROVED** [review]

**Pull request merged**